reproject/high_level.py

```diff
@@ -43,6 +43,7 @@ def reproject(input_data, output_projection, shape_out=None, projection_type='bi
             * 'biquadratic'
             * 'bicubic'
             * 'flux-conserving'
+            * 'deforest' (C.E. DeForest, Solar Physics 2004)
```

Comment:
In the PR comment it says DeForest 2003, and here 2004...  Is it this one:
DOI: [10.1023/B:SOLA.0000021743.24248.b0](http://dx.doi.org/10.1023/B:SOLA.0000021743.24248.b0)
Maybe adding the DOI number to the docstring would help for people to find the paper?

@astrofrog great job by the way!